From f9c104055132d2b8b333cbf2873c4a06e50d9596 Mon Sep 17 00:00:00 2001
From: Juergen Repp <juergen_repp@web.de>
Date: Thu, 23 May 2024 11:40:14 +0200
Subject: [PATCH 02/18] tpm2_evictcontrol.c: Fix segfault for output of handle
 ESYS_TR_NONE

If the -o parameter was used when a persistent handle should be
evicted a segfault did occur.
Now the object is evicted an a warning is displayed.

Signed-off-by: Juergen Repp <juergen_repp@web.de>
---
 tools/tpm2_evictcontrol.c | 6 +++++-
 1 file changed, 5 insertions(+), 1 deletion(-)

diff --git a/tools/tpm2_evictcontrol.c b/tools/tpm2_evictcontrol.c
index 898b4ca7..73d477bc 100644
--- a/tools/tpm2_evictcontrol.c
+++ b/tools/tpm2_evictcontrol.c
@@ -102,7 +102,11 @@ static tool_rc process_output(ESYS_CONTEXT *ectx) {
 
     tool_rc tmp_rc = tool_rc_success;
     if (ctx.output_arg) {
-        tmp_rc = files_save_ESYS_TR(ectx, ctx.out_tr, ctx.output_arg);
+        if (ctx.out_tr == ESYS_TR_NONE) {
+            LOG_WARN("The parameter --output can't be used if a persistent object will be evicted");
+        } else {
+            tmp_rc = files_save_ESYS_TR(ectx, ctx.out_tr, ctx.output_arg);
+        }
     }
 
     if (!is_evicted) {
-- 
2.49.0

